client/repoutils.c

```diff
@@ -229,27 +226,62 @@ TDNFRepoGetRpmCacheDir(
     goto cleanup;
 }
 
+/* remove the repo top level cache dir */
+uint32_t
+TDNFRepoRemoveCacheDir(
+    PTDNF pTdnf,
+    PTDNF_REPO_DATA pRepo
+    )
+{
+    uint32_t dwError = 0;
+    char* pszRepoCacheDir = NULL;
+
+    if(!pTdnf || !pRepo)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    dwError = TDNFGetCachePath(pTdnf, pRepo,
+                               NULL, NULL,
+                               &pszRepoCacheDir);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    if (rmdir(pszRepoCacheDir) != 0)
+    {
+        /* ignore ENOTEMPTY, let's keep the dir if it's not empty */
```

Comment:
I think we should not ignore not empty, it might lead to partial clean ups

Reply:
It only makes sense to remove the directory when it's empty. And this *is* executed when we do partial cleanups (something other than `all`), so leaving stuff behind is intended.

Maybe we can print a warning when the arg is `all` but the dir is not empty. Because this means either that we missed something, or the user put some files there.